templates/assets/js/navigation.js

```diff
@@ -130,20 +129,46 @@ app.navigation = (function () {
                                 break;
                             case 'chart':
                                 app.chart.toggleChart();
+                                self.displayAllScenarios(scenarios);
                                 break;
                             case 'steps':
-                                self.displayAllScenarios(scenarios);
-                                self.toggleSteps(steps);
+                                if (document.querySelector('.error_btn').classList.contains('active')) {
```

Comment:
@maciejkamela mysle, ze powinienes oddzielic widok od logiki i np ustawic sobie zmienna 
currentView ktora przyjmuje nastepujace wartosci (all, passed, failed)

i wtedy klikniecie  w button error ustawi currentView na failed
kilkniecie w button passed ustawi currentView na passed itd.
i wtedy w tym ifie sprawdzacz if (currentView === 'failed') zrob to i to

Wtedy logika jest oddzielona od widoku. Wyobraz sobie ze bedziesz chcial zmienic klase z active na activeBtn i wtedy aplikacja Ci sie sypnie